### DISCUSSION TITLE
a bit OT: I installed SCO Xenix on 86Box

### DISCUSSION THREAD

**Discussion by Randrianasulu** [jbruchon/elks#1574] [General]
from https://winworldpc.com/product/xenix/286

using 286 config
[86box-xenix.txt](https://github.com/jbruchon/elks/files/11068598/86box-xenix.txt)

To my surprise it worked OK, while initially I had trouble  writing to floppy represented by host file owned by root :)

I can create 1.2 floppy image on host, mkfs it inside emulator, mount it there, copy files on it, unmount /dev/fd0 , eject disk and mount it on Linux host:

dd if=/dev/zero of=floppy_transfer.img bs=1k count=1200
[do things in 86Box]
 mount floppy_transfer.img /mnt/test -t sysv

file /mnt/test/cc
/mnt/test/cc: Microsoft a.out separate pure segmented fixed-stack word-swapped V2.3 V3.0 86 small model executable

file /mnt/test/vi
/mnt/test/vi: sticky Microsoft a.out separate pure segmented fixed-stack word-swapped V2.3 V3.0 286 small model executable Large Text

So, yeah, apparently large code/data model useful not just for compilers (I spotted MASM there as well) but for editors, compressors (gzip), pagers (less) ...

**Comment by Randrianasulu** [comment]
Currently reading old pists archived at

https://www.linux.co.cr/unix-source-code/index.html

Already copied dosread.c for some testing on Xenix (found whole page via google while looking for xenix and dosread). People already discussed 64k segments, with some strongly-worded opinions :) But also there was note pdp11 also might have them, so it was not completely unheard about ..

World moved on, but swap (virtual memory) still refuses to die after all those decades.

Also looked yesterday on Xenix install thread, something interesting was posted there :)

https://forum.vcfed.org/index.php?threads/my-latest-xenix-build.49617/

**Comment by ghaerr** [comment]
@Randrianasulu: Thanks for the links and comments.

One of the big differences between what we're doing at ELKS compared to Xenix is that we've got complete source code for all development tools, as well as the kernel, C library and applications. So it can move forward in ways we can't with Xenix unless the source and tools were open sourced. It was interesting to read that SCO's assembler was actually Microsoft's MASM 3.0 renamed.

**Comment by Randrianasulu** [comment]
@ghaerr yeah, while right now I am learning hard way why ~1987 compiler is not exactly fun, even if it supports big memory models :) it can't compile even this unproto software fully, same for i/o benchmark and dosread command ...

Interesting aside - it seems at least Xenix/286 moved in direction of having three separate *.a sets in lib, for each memory model.
